packages/ag-charts-community/src/module/module.ts

```diff
@@ -56,48 +56,50 @@ class ModuleRegistry {
     }
 
     *byType<T extends Module>(...types: Module['type'][]): Generator<T> {
-        for (const module of this.modules) {
-            if (types.includes(module.type) && this.dependencies.has(module.optionsKey)) {
-                yield module as T;
-            }
-        }
+        const { dependents } = this;
 
-        for (const module of this.modules) {
-            if (
-                types.includes(module.type) &&
-                !this.dependencies.has(module.optionsKey) &&
-                !this.dependents.has(module.optionsKey)
-            ) {
-                yield module as T;
-            }
-        }
+        const yielded = new Set();
+
+        let count = 0;
+        const maxCount = 3;
+
+        const modulesByType = this.modules.filter((module) => types.includes(module.type));
+
+        do {
```

Comment:
Might be worth adding a comment here to explain what is going on - my understanding is this is trying to take an iterative approach to resolving dependencies rather than a recursive approach, with a max iteration limit of 3?

Alternatively a recursive approach is fine, the only thing you need to do is pass down context of which modules have already been considered so you can detect loops and fail-fast?